core/unit_test/cuda/TestCuda_InterOp_GraphMultiGPU.cpp

```diff
@@ -0,0 +1,115 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include "Kokkos_Core.hpp"
+#include "Kokkos_Graph.hpp"
+#include "TestCuda_Category.hpp"
+
+namespace {
+
+class TEST_CATEGORY_FIXTURE(multi_gpu) : public ::testing::Test {
+ public:
+  using graph_t          = Kokkos::Experimental::Graph<Kokkos::Cuda>;
+  using graph_node_ref_t = Kokkos::Experimental::GraphNodeRef<Kokkos::Cuda>;
+
+ public:
+  void SetUp() override {
+    KOKKOS_IMPL_CUDA_SAFE_CALL(cudaGetDeviceCount(&num_devices));
+
+    if (num_devices < 2)
+      GTEST_SKIP()
+          << "Skipping multi-gpu testing since current machine only has "
+          << num_devices << " GPU.";
+
+    execs.reserve(num_devices);
+
+    for (int idevice = 0; idevice < num_devices; ++idevice) {
+      KOKKOS_IMPL_CUDA_SAFE_CALL(cudaSetDevice(idevice));
+      cudaStream_t stream = nullptr;
+      KOKKOS_IMPL_CUDA_SAFE_CALL(cudaStreamCreate(&stream));
+      execs.push_back(Kokkos::Cuda(stream));
+    }
+  }
+
+  void TearDown() override {
+    for (const auto& exec : execs) {
+      KOKKOS_IMPL_CUDA_SAFE_CALL(cudaStreamDestroy(exec.cuda_stream()));
```

Comment:
This is semantically wrong.
You need to delete the Kokkos exec spaces first and then the CUDA streams.